src/javascript/binary/static/profittable/profittable.ui.js

```diff
@@ -0,0 +1,122 @@
+
+var ProfitTableUI = (function(){
+    "use strict";
+
+    var profitTableID = "profit-table";
+    var cols = ["buy-date", "ref", "contract", "buy-price", "sell-date", "sell-price", "pl"];
+    var header = ["Purchase Date", "Ref.", "Contract", "Purchase Price", "Sale Date", "Sale Price", "Profit/Loss"];
+    var footer = ["Total Profit/Loss", "", "", "", "", "", ""];
+
+    function createEmptyTable(){
+        function mergeRows(start, end, $row){
+            var texts = $row.children().map(function (i, obj) {
+                if (i >= start && i <= end) {
+                    var text = obj.textContent;
+                    if (i !== start) {
+                        obj.remove();
+                    }
+                    return text;
+                }
+            }).toArray().join(" ");
+
+            $row.
+                children().
+                eq(start).
+                text(texts).
+                attr("colspan", end - start + 1);
+        }
+
+        var localizedHeader = header.map(function(t){return text.localize(t);});
+
+        var data = [];
+        var metadata = {
+            cols: cols,
+            id: profitTableID
+        };
+        var $tableContainer = Table.createFlexTable(data, metadata, localizedHeader, footer);
+
+        var $pltotal = $tableContainer.
+            children("table").
+            children("tfoot").
+            children("tr").
+            attr("id", "pl-day-total");
+
+        mergeRows(0, 5, $pltotal);
+
+        return $tableContainer;
+    }
+
+    function updateProfitTable(transactions){
+        Table.appendTableBody(profitTableID, transactions, createProfitTableRow);
+        updateFooter(transactions);
+    }
+
+    function updateFooter(transactions){
+        var accTotal = document.querySelector("#pl-day-total > .pl").textContent;
+        accTotal = parseFloat(accTotal);
+        if (isNaN(accTotal)) {
+            accTotal = 0;
+        }
+
+        var currentTotal = transactions.reduce(function(previous, current){
+            var buyPrice = Number(parseFloat(current["buy_price"])).toFixed(2);
+            var sellPrice = Number(parseFloat(current["sell_price"])).toFixed(2);
+            var pl = sellPrice - buyPrice;
+            return previous + pl;
+        }, 0);
+
+        var total = accTotal + currentTotal;
+
+        $("#pl-day-total > .pl").text(Number(total).toFixed(2));
+
+        var subTotalType = (total >= 0 ) ? "profit" : "loss";
+        $("#pl-day-total > .pl").removeClass("profit").removeClass("loss");
+        $("#pl-day-total > .pl").addClass(subTotalType);
+    }
+
+    function createProfitTableRow(transaction){
+        var buyDate = moment.
+            utc(transaction["purchase_time"] * 1000).
+            locale("en").
```

Comment:
Should we be considering the current language for localization of the date?
I don't think we are doing it now, but may be we should?